src/main/java/picard/sam/RevertSam.java

```diff
@@ -374,6 +383,21 @@ public void revertSamRecord(final SAMRecord rec) {
                     }
                 }
 
+                // Remove records that have the SAM SAMString (*** SLOW ***)
+                if (DEDUPLICATE_RECORDS) {
+                    final Iterator<SAMRecord> iter = recs.iterator();
+                    final Set<String> samStrings = new HashSet<>();
```

Comment:
1. I'm unclear why this is better than just requiring that there be only a single unpaired/R1/R2 in the collection.  That's tested below, so even if they are not identical, it'll still fail if there are, e.g. two R1s in the collection.
2. I'd be a little concerned that if the extended attributes are in a different order they might generate different strings while otherwise being identical.  Is `SAMRecord.equals()` not trustworthy for some reason?